src/core/tag/mixins/pure.ts

```diff
@@ -1,161 +1,161 @@
 import Tag from '..';
 import * as utils from '../../utils';
 import Phase from '../phase';
-import buildProps from '../props';
+import Props from '../props';
 import TagUtils from '../utils';
 
 export const RE_RENDER_MESSAGE = '%ctag is preparing to re-render:';
 
-export default function pureMixin(this: Tag) {
-  this.shouldUpdate = shouldUpdate(this);
-}
+namespace Pure {
+  export function pureMixin(this: Tag) {
+    this.shouldUpdate = Pure.shouldUpdate(this);
+  }
 
-export function shouldUpdate(tag: Tag) {
-  let prevState = null;
-  let prevProps = null;
-  let prevAliases = null;
-  const updatePrev = () => {
-    prevState = tag.state;
-    prevProps = tag.props;
-    prevAliases = extractLocalAliases(tag);
-  };
-
-  tag.one(Phase.MOUNT, updatePrev);
-  tag.on(Phase.UPDATED, updatePrev);
-
-  return (stateChange: object, nextOpts: object) => {
-    const nextProps = buildProps(tag, nextOpts);
-    const nextState = stateChange != null ? { ...tag.state, ...stateChange['state'] } : tag.state;
-    const nextAliases = resolveAliases(tag);
-    const propsUpdated = !shallowEquals(prevProps, nextProps);
-    const stateUpdated = !shallowEquals(prevState, nextState);
-    const aliasesUpdated = !shallowEquals(prevAliases, nextAliases);
-
-    if (aliasesUpdated || stateUpdated || propsUpdated) {
-      if (TagUtils.isDebug(tag.config)) {
-        let message = RE_RENDER_MESSAGE;
-        console.dirxml(tag.root);
-        switch (true) {
-          case aliasesUpdated:
-            message += ' %can alias dependency was updated';
-            break;
-          case stateUpdated && stateChange == null:
-            message += ' %can update was forced';
-            break;
-          case stateUpdated && stateChange != null:
-            message += ' %ca state change occurred';
-            break;
-          case propsUpdated:
-            message += ' %cit received new props';
-            break;
+  export function shouldUpdate(tag: Tag) {
+    let prevState = null;
+    let prevProps = null;
+    let prevAliases = null;
+    const updatePrev = () => {
+      prevState = tag.state;
+      prevProps = tag.props;
+      prevAliases = Pure.extractLocalAliases(tag);
+    };
+
+    tag.one(Phase.MOUNT, updatePrev);
+    tag.on(Phase.UPDATED, updatePrev);
+
+    return (stateChange: object | true, nextOpts: object) => {
+      const nextProps = Props.buildProps(tag, nextOpts);
+      const forceUpdate = stateChange === true;
```

Comment:
`stateChange === true || !('state' in stateChange)`

Reply:
no